Assets/Scripts/Pathfinding/Pathfinder.cs

```diff
@@ -0,0 +1,226 @@
+                int maxY = goalTile.Y + 1;
+
+                return tile => (
+                    tile.X >= minX && tile.X <= maxX &&
+                    tile.Y >= minY && tile.Y <= maxY);
+            }
+            else
+            {
+                return tile => goalTile == tile;
+            }
+        }
+
+        public static PathfindingHeuristic DefaultDistanceHeuristic(Tile goalTile)
+        {
+            return ManhattanDistance(goalTile);
+        }
+
+        /// <summary>
+        /// ManhattanDistance measurement.
+        /// </summary>
+        /// <param name="goalTile">Goal tile.</param>
+        public static PathfindingHeuristic ManhattanDistance(Tile goalTile)
+        {
+            return tile => Mathf.Abs(tile.X - goalTile.X) + Mathf.Abs(tile.Y - goalTile.Y);
+        }
+
+        public static GoalEvaluator GoalInventoryEvaluator(string[] objectTypes, bool canTakeFromStockpile = true)
+        {
+            return tile => InventoryManager.InventoryCanBePickedUp(tile.Inventory, canTakeFromStockpile) && objectTypes.Contains(tile.Inventory.Type);
+        }
+
+        /// <summary>
+        /// Evaluates if the goal is an inventory of the right type.
+        /// </summary>
+        /// <param name="objectType">Inventory's object type.</param>
+        /// <param name="canTakeFromStockpile">If set to <c>true</c> can take from stockpile.</param>
+        public static GoalEvaluator GoalInventoryEvaluator(string objectType, bool canTakeFromStockpile = true)
+        {
+            return tile => InventoryManager.InventoryCanBePickedUp(tile.Inventory, canTakeFromStockpile) && objectType == tile.Inventory.Type;
+        }
+
+        /// <summary>
+        /// Evaluates if the goal is a furniture of the right type.
+        /// </summary>
+        /// <param name="objectType">Inventory's object type.</param>
+        public static GoalEvaluator GoalFurnitureEvaluator(string objectType)
+        {
+            return current => current.Furniture != null && current.Furniture.Type == objectType;
+        }
+
+        /// <summary>
+        /// Evaluates if it is an appropriate place to dump objectType of amount
+        /// </summary>
+        public static GoalEvaluator GoalCanFitInventoryEvaluator(string objectType, int amount)
+        {
+            return tile => tile.Type == TileType.Floor && (
+                tile.Inventory == null ||
+                tile.Inventory.Type == objectType && (tile.Inventory.StackSize + amount) <= tile.Inventory.MaxStackSize);
+        }
+
+        /// <summary>
+        /// Dijkstra's algorithm.
+        /// </summary>
+        public static PathfindingHeuristic DijkstraDistance()
+        {
+            return tile => 0f;
+        }
+
+        private static void DebugLog(string message, params object[] par)
+        {
+            Debug.ULogChannel("Pathfinding", message, par);
+        }
+
+        private static void DebugLogIf(bool condition, string message, params object[] par)
```

Comment:
My way to deal with our style guide not letting me write if-statements without bracket. It would make sense to expand and make available more generally.